### PR TITLE
C sharp rascal mcs wrapper

### PR DESCRIPTION
**Pull request by jones-gareth**
#### What does this implement/fix? Explain your changes.

Add Rascal MCES to the C# swig wrapper

#### Any other comments?

The Rascal code uses `std::shared_ptr` whereas the wrappers use `boost::shared_ptr`, which complicates the interface file.  I might be better for Rascal to use `boost::shared_ptr`.

### CODE REVIEW

**Comment by greglandrum** [comment]
@jones-gareth I just did a PR against your branch to also support Java wrappers

**Review by greglandrum — APPROVED** [review]
LGTM

**Comment by greglandrum** [comment]
@ptosco: do you also want to take a look at this one?

File: Code/JavaWrappers/RascalMCES.i [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		}
+	}
+%}
+
+
+%extend RDKit::RascalMCES::RascalApp {
+    static std::vector<std::vector<unsigned int> > RascalCluster(const std::vector<boost::shared_ptr<RDKit::ROMol> >& mols, const RDKit::RascalMCES::RascalClusterOptions& clusterOptions=RascalClusterOptions()) {
+		std::vector<std::shared_ptr<RDKit::ROMol> > rascalMolecules;
+		for (auto molIn: mols) {
+			rascalMolecules.push_back(std::make_shared<RDKit::ROMol>(*molIn));
```
Comment (ptosco) [inline Code/JavaWrappers/RascalMCES.i:58]:
```suggestion
			rascalMolecules.emplace_back(new RDKit::ROMol(*molIn));
```

File: Code/JavaWrappers/RascalMCES.i [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		}
+		return RDKit::RascalMCES::rascalCluster(rascalMolecules, clusterOptions);
+	}
+    static std::vector<std::vector<unsigned int> > RascalButinaCluster(const std::vector<boost::shared_ptr<RDKit::ROMol> >& mols, const RDKit::RascalMCES::RascalClusterOptions& clusterOptions=RascalClusterOptions()) {
+		std::vector<std::shared_ptr<RDKit::ROMol> > rascalMolecules;
+		for (auto molIn: mols) {
+			rascalMolecules.push_back(std::make_shared<RDKit::ROMol>(*molIn));
```
Comment (ptosco) [inline Code/JavaWrappers/RascalMCES.i:65]:
```suggestion
			rascalMolecules.emplace_back(new RDKit::ROMol(*molIn));
```

**Review by ptosco — APPROVED** [review]
LGTM. I propose two very minor simplifications - feel free to ignore them.

**Comment by jones-gareth** [comment]
Committed @ptosco 's changes

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**